website/content/docs/job-specification/consul.mdx

```diff
@@ -0,0 +1,236 @@
+`consul.task_identity` and `consul.service_identity`. Refer to [Configuring Consul
+Authentication][] for more details.
+
+<Warning>
+Starting in Nomad 1.9, the fallback options to use the -consul-token flag when
+submitting a job, the agent's consul.token configuration, or the
+CONSUL_HTTP_TOKEN environment variable, will be removed.  This means service and
+template blocks will not be able to use the agent's Consul token or one provided
+by the job submitter. You should be prepared to migrate to the Workload Identity
+workflow for Consul and Vault before upgrading to Nomad 1.9. Refer to <a
+href="/nomad/docs/integrations/consul-integration#migrating-to-using-workload-identity-with-consul">Migrating
+to Using Workload Identity with Consul</a>
+</Warning>
+
+### Access to Token
+
+The Nomad client will make the Consul token available to the task by writing it
+to the secret directory at `secrets/consul_token` and by injecting a
+`CONSUL_TOKEN` environment variable in the task. If the Nomad cluster is
+[configured][config_consul_namespace] to use [Consul Namespaces][], a
+`CONSUL_NAMESPACE` environment variable will be injected whenever `CONSUL_TOKEN`
+is set.
+
+The [`template`][template] block can use the Consul token as well.
+
+### `consul` Parameters
+
+- `cluster` `(string: "default")` <EnterpriseAlert inline/> - Specifies the
+  Consul cluster to use. The Nomad client will retrieve a Consul token from the
+  cluster configured in the agent configuration with the same
+  [`consul.name`][]. In Nomad Community Edition, this field is ignored.
+
+- `namespace` `(string: "")` <EnterpriseAlert inline/> - The Consul namespace in
+  which group and task-level services within the group will be registered. Use
+  of `template` to access Consul KV will read from the specified Consul
+  namespace.  Specifying `namespace` takes precedence over the
+  [`-consul-namespace`][flag_consul_namespace] command line argument in `job run`.
+
+## `consul` Examples
+
+The following examples only show the `consul` blocks or other relevant
+blocks. Remember that the `consul` block is only valid in the placements listed
+above.
+
+### Consul Token for Templates
+
+This example tells the Nomad client to obtain a Consul token using the task's
+Workload Identity. The token is available to the task via the canonical
+environment variable `CONSUL_TOKEN` and written to disk at
+`secrets/consul_token`. The `template` block will use the same Consul token.
+
+```hcl
+job "example" {
+  group "app" {
+    task "web" {
+      identity {
+        name = "consul_default"
+      }
+
+      template {
+        data        = "APP_NAME={{key \"app/name\"}}"
+        destination = "local/config.txt"
+      }
+    }
+  }
+}
+```
+
+The following example has the same behavior as the example above, but assumes
+that the server has a `consul.task_identity` block configured. Note that both
+the `consul` and `identity` blocks can be omitted in this case.
```

Comment:
Ah interesting, I think an empty `vault` block is still required in the Vault side. 

How is it decided that a Consul token is required for this task?

Reply:
Ah right. Job files already have the `vault` blocks in the expected places 👍